### PR TITLE
Update ServiceBus module to use topics and topics authorizationRules

### LINKED ISSUE #1027: Add topics child resource to Service Bus module
## Description

Update Service Bus module to include `Microsoft.ServiceBus/namespaces/topics` and `Microsoft.ServiceBus/namespaces/topics/authorizationRules` consistently with queues

### PR DESCRIPTION
**Pull request by eriqua**
# Change

Integrating `Microsoft.ServiceBus/namespaces/topics` and `Microsoft.ServiceBus/namespaces/topics/authorizationRules`

[ServiceBus: Namespaces](https://github.com/Azure/ResourceModules/actions/workflows/ms.servicebus.namespaces.yml)

Closes #1027

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: arm/Microsoft.ServiceBus/namespaces/topics/deploy.bicep [hunk 1]
```diff
@@ -0,0 +1,147 @@
+
```
Comment (MrMCake) [inline arm/Microsoft.ServiceBus/namespaces/topics/deploy.bicep:1]: 2 redundant empty lines

File: arm/Microsoft.ServiceBus/namespaces/topics/readme.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#  `[]`
```
Comment (MrMCake) [inline arm/Microsoft.ServiceBus/namespaces/topics/readme.md:1]: This readme's header must be updated

File: arm/Microsoft.ServiceBus/namespaces/topics/readme.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+| `namespaceName` | string |  |  | Required. Name of the parent Service Bus Namespace for the Service Bus Topic. |
+| `requiresDuplicateDetection` | bool |  |  | Optional. A value indicating if this topic requires duplicate detection. |
+| `roleAssignments` | array | `[]` |  | Optional. Array of role assignment objects that contain the 'roleDefinitionIdOrName' and 'principalId' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: '/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11' |
+| `status` | string | `Active` | `[Active, Disabled, Restoring, SendDisabled, ReceiveDisabled, Creating, Deleting, Renaming, Unknown]` | Optional. Enumerates the possible values for the status of a messaging entity. - Active, Disabled, Restoring, SendDisabled, ReceiveDisabled, Creating, Deleting, Renaming, Unknown |
+| `supportOrdering` | bool |  |  | Optional. Value that indicates whether the topic supports ordering. |
+
+### Parameter Usage: `<ParameterPlaceholder>`
```
Comment (MrMCake) [inline arm/Microsoft.ServiceBus/namespaces/topics/readme.md:37]: Remove

File: arm/Microsoft.ServiceBus/namespaces/topics/deploy.bicep (hunk 1 above)
Comment (eriqua) [reply arm/Microsoft.ServiceBus/namespaces/topics/deploy.bicep]: done

File: arm/Microsoft.ServiceBus/namespaces/topics/readme.md (hunk 2 above)
Comment (eriqua) [reply arm/Microsoft.ServiceBus/namespaces/topics/readme.md]: done

File: arm/Microsoft.ServiceBus/namespaces/topics/readme.md (hunk 3 above)
Comment (eriqua) [reply arm/Microsoft.ServiceBus/namespaces/topics/readme.md]: done

**Pull request merged**